src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -101,7 +101,9 @@ public override Task WriteAsync(byte[] buffer, int offset, int count, Cancellati
 
         public ValueTask ShutdownWriteCompleted(CancellationToken cancellationToken = default) => _provider.ShutdownWriteCompleted(cancellationToken);
 
-        public void Shutdown() => _provider.Shutdown();
+        public override void CompleteWrites() => _provider.CompleteWrites();
```

Comment:
Why are we replacing `Shutdown`? I thought this is a replacement for `ShutdownWriteCompleted`? Or did I completely miss the whole point of this change somehow?

Reply:
ShutdownWriteCompleted will become something like `Task Completed { get; }`, which is what its current behavior is.

Reply:
I thought that original `ShutdownWriteCompleted` means "I'm finished writing you can start sending response", which I thought is the meaning of the new `CompleteWrites`?
I also thought that original `Shutdown` is the final step of the communication, i.e.: after the response has been sent.

I don't know anything about `Task Completed { get; }`, I don't think we got that far with the QUIC API reviews, but if I understand the meaning that should correspond to `Shutdown`.

Reply:
`Shutdown` is "shutdown writes, send EOF to peer".

`ShutdownWriteCompleted` was "wait for peer to acknowledge that shutdown" which we found to be mostly useless so I believe we changed it to mean "wait for peer to acknowledge that shutdown AND to shutdown their side too". It has a terrible name. This is what I'd like to see changed to a `Task Completed { get; }`.

Reply:
`ShutdownWriteCompleted` actually remained for now, I've added new `ShutdownCompleted` alongside... But I agree `ShutdownWriteCompleted` doesn't have any use for us anymore, so it can just be removed anytime

Reply:
I was completely bamboozled by the naming and made assumption based on that.